### ISSUE TITLE
TextField doesn't properly create a new TextEditingController on didUpdateWidget

### ISSUE THREAD

**Issue by paolosoares** [flutter/flutter#12153]
In TextEditingController's [didUpdateWidget](https://github.com/flutter/flutter/blob/ca7d2d23cf601d42c1e1354355116c0c96f7fdb2/packages/flutter/lib/src/material/text_field.dart#L216) method:

```dart
  @override
  void didUpdateWidget(TextField oldWidget) {
    super.didUpdateWidget(oldWidget);
    if (widget.controller == null && oldWidget.controller != null)
      _controller == new TextEditingController.fromValue(oldWidget.controller.value);
    else if (widget.controller != null && oldWidget.controller == null)
      _controller = null;
  }
```
When `widget.controller` is `null` and `oldWidget.controller` isn't, `_controller` gets assigned instead of copied. Specifically, this:

```dart
_controller == new TextEditingController.fromValue(oldWidget.controller.value);
```

should be

```dart
_controller = new TextEditingController.fromValue(oldWidget.controller.value);
```

**Comment by Hixie** [comment]
(Filed https://github.com/dart-lang/sdk/issues/30793 for a lint that would have caught this.)

cc @xster, @cbracken

**Comment by cbracken** [comment]
Thanks! Fixed by #12190.
